frontend/packages/dev-console/src/components/add/AddCardSection.tsx

```diff
@@ -23,22 +25,38 @@ const AddCardsSection: React.FC<AddCardsSectionProps> = ({
   namespace,
   addActionExtensions,
   addActionGroupExtensions,
+  loading,
 }) => {
   const filteredAddActionExtensions: ResolvedExtension<AddAction>[] = filterNamespaceScopedUrl(
     namespace,
     useAccessFilterExtensions(namespace, addActionExtensions),
   );
 
-  const sortedActionGroup: LoadedExtension<AddActionGroup>[] = getSortedExtensionItems<
-    LoadedExtension<AddActionGroup>
-  >(addActionGroupExtensions);
+  const isLoading: boolean = loading || filteredAddActionExtensions.length === 0;
 
-  const addGroups: AddGroup[] = getAddGroups(filteredAddActionExtensions, sortedActionGroup);
+  const getAddCards = (): React.ReactElement[] => {
+    if (isLoading) {
+      return [];
+    }
+    const sortedActionGroup: LoadedExtension<AddActionGroup>[] = getSortedExtensionItems<
+      LoadedExtension<AddActionGroup>
+    >(addActionGroupExtensions);
 
-  const addCards: React.ReactElement[] = addGroups.map(({ id, name, items }) => (
-    <AddCard key={id} title={name} items={items} namespace={namespace} />
-  ));
+    const addGroups: AddGroup[] = getAddGroups(filteredAddActionExtensions, sortedActionGroup);
 
-  return <MasonryLayout columnWidth={COLUMN_WIDTH}>{addCards}</MasonryLayout>;
+    return addGroups.map(({ id, name, items }) => (
+      <AddCard key={id} title={name} items={items} namespace={namespace} />
```

Comment:
I thought we added a `data-test-id` to each AddCard already in the other PR. Maybe we lost this. Can you add this `data-test-id` as part of this PR?